### DISCUSSION TITLE
Framework Choice

### DISCUSSION THREAD

**Discussion by NobleTyler** [codetocloudinc/code-to-cloud-home#3] [Polls]
Hey team,

As you can see, we’re kicking off our event site using elements from Project Hoverboard. Nothing is set in stone yet, so I’d love to gather your thoughts. Do you have any preferences for the framework or technology we should use? Is there a tool or framework you’re experienced with that you’d like to bring to the project?

Also, a heads-up from @kevinevans1: there are some security issues in this repo, so if anyone is interested, there’s a good opportunity to dive into **cybersecurity**. As well there is a cloud function in there as well we need to figure out. So for those interested in cloud functions there's good fun to be had!

That said I'm open to hearing any ideas you may have on what you'd like to see. Be it using a framework like [Expo](https://expo.dev/) or [Remix](https://remix.run/), or sticking with Hoverboard which seems to be a good fit for our use case. Let me know what you think!

Looking forward to hearing your thoughts!

**Comment by victoriacheng15** [comment]
I looked into the hoverboard, but there doesn’t seem to be much information available about it. I'm concerned about relying on it, especially if we need support or troubleshooting in the future. Maybe we could consider choosing a framework with an active community and more support resources?

**Comment by haywirebanana** [comment]
Looking at Hoverboard, I understand that majority of the work is done. The problem I see in this is the effort into understanding and updating outdated code and even documentation. Starting from scratch may allow us to be in a consensus for any technical decisions done to it. 
(This may allow us to learn more from this experience as well)

**Comment by NobleTyler** [comment]
Love the comments @haywirebanana and @victoriacheng15. I think you're right in that troubleshooting particularly as it comes to cybersecurity and cloud functions could prove trickier than the build itself.

I'd like to propose using [Astro](https://astro.build/) to build out our site. It has the CMS integrations we need. Is most importantly fast and allows us to use a React Syntax to build pages so we have some familiarity there as well.

I'm liking the options which are ServerSide rendered as they'll help with SEO which will be key in getting people to find out about CodeToCloud. What are everyone's thoughts on Astro?

**Reply by victoriacheng15** [reply]
Astro is a good choice and great for blogs. They release updates very often with an active community.

I built my site with astro and love it so far!

**Comment by nataliey16** [comment]
I think Project Hoverboard covers many of the user story needs, like quick deployment, cost-efficiency, reliability, responsive design, and social media integration. However, it lacks flexibility and scalability, especially if we switch to Azure, which could bring challenges due to limited documentation (similarly to what is mentioned above). If we build from scratch, we have full customization, scalability, flexibility on the tech stack flexibility, and integrating third-part resources. This might make it a better fit for meeting long-term goals. With the right planning, i think we can meet all the requirements of the user story!
